src/utils/utils.spec.ts

```diff
@@ -0,0 +1,199 @@
+import Int from '@polkadot/types/codec/Int';
+import UInt from '@polkadot/types/codec/UInt';
+
+import {
+	kusamaRegistry,
+	PRE_SANITIZED_BALANCE_LOCK,
+	PRE_SANITIZED_OPTION_VESTING_INFO,
+	PRE_SANITIZED_RUNTIME_DISPATCH_INFO,
+	PRE_SANITIZED_STAKING_RESPONSE,
+} from './test_util';
+import { sanitizeNumbers } from './utils';
+
+describe('sanitizeNumbers', () => {
+	it('does not affect non-numbers', () => {
+		expect(sanitizeNumbers('Hello world')).toBe('Hello world');
+	});
+
+	it('does not convert plain hexadecimal', () => {
+		expect(sanitizeNumbers('40C0A7')).toBe('40C0A7');
+		expect(sanitizeNumbers('0x40C0A7')).toBe('0x40C0A7');
+	});
+
+	it('converts U32 Int and Uint', () => {
+		const intTen = new Int(kusamaRegistry, 10);
+		expect(sanitizeNumbers(intTen)).toBe('10');
+
+		const intPaddedHex = new Int(
+			kusamaRegistry,
+			'0x000000000000000004fe9f24a6a9c00'
+		);
+		expect(sanitizeNumbers(intPaddedHex)).toBe('22493750000000000');
+
+		const uIntTen = new UInt(kusamaRegistry, 10);
+		expect(sanitizeNumbers(uIntTen)).toBe('10');
+
+		const uIntPaddedHex = new UInt(
+			kusamaRegistry,
+			'0x000000000000000004fe9f24a6a9c00'
+		);
+		expect(sanitizeNumbers(uIntPaddedHex)).toBe('22493750000000000');
+
+		const u32Max = kusamaRegistry.createType('u32', '0x7FFFFFFF');
+		expect(sanitizeNumbers(u32Max)).toBe('2147483647');
```

Comment:
Why not use the first bit? It's unsigned.
https://doc.rust-lang.org/std/primitive.u32.html#associatedconstant.MAX